projects/python-client/src/datapane/client/config.py

```diff
@@ -105,39 +112,47 @@ def load(cls, env: str = "default") -> "Config":
         return config
 
     @classmethod
-    def create_default(cls, env: str, config_f: Path) -> None:
+    def create_default(cls) -> None:
         """Create an default config file"""
         # create default file
         _config = Config()
-        _config.save(env)
-        log.info(f"Created config file at {config_f}")
+        _config.save()
+        log.info(f"Created config file at {CONFIG_PATH}")
 
-    def save(self, env: t.Optional[str] = None):
-        assert env or self._path
-
-        if env:
-            # NOTE - type checker doesn't like us setting class variables on instance
-            # don't think this will cause issues
-            self._env = env
-            self._path = self.get_config_file(env)
-
-        with self._path.open("w") as f:
-            config_dictionary = dc.asdict(self)
-            del config_dictionary["_path"]
-            del config_dictionary["_env"]
-            yaml.safe_dump(config_dictionary, f)
+    def save(self):
+        data = dc.asdict(self)
+        parser = configparser.ConfigParser()
+        parser[CONFIG_SECTION] = data
+        with CONFIG_PATH.open("w") as file:
+            parser.write(file)
 
     def remove(self):
-        self._path.unlink()
+        CONFIG_PATH.unlink()
 
-    @staticmethod
-    def get_config_file(env: str) -> Path:
-        return APP_DIR / f"{env}.yaml"
+    @classmethod
+    def ensure_config_file(cls) -> None:
+        """
+        Ensure the config file exists at CONFIG_PATH
+        """
+        if CONFIG_PATH.exists():
+            return
+
+        # Look for YAML from Version 5 and earlier
+        yaml_path = LEGACY_CONFIG_PATH
+        if yaml_path.exists():
+            # Migrate
+            log.info(f"Found legacy config file at {yaml_path}")
+            CONFIG_PATH.write_text(f"[{CONFIG_SECTION}]\n" + yaml_path.read_text())
+            yaml_path.unlink()
+            log.info(f"Successfully migrated to {CONFIG_PATH}")
```

Comment:
This converts a single yaml file, for the `default` env, but doesn't convert the others. Is that the intended behaviour - do we need to convert the other config files the user may have?

Reply:
My thinking was if we're removing multiple envs because we believe most users will only have the one default env, if we just update that it should work for the majority of users. We may as well then leave the old yaml files there so we don't accidentally destroy tokens.

I did think about adding an undocumented os.env check to let us still swap config files - if we go down that road then converting other yaml files would make sense, but probably best to upgrade them one by one when they are first loaded, rather than seek out everything in the conf dir?

Reply:
I think we can ignore the other env files - I don't think any user will ever have created them (I was the only person, that I know of, that even used them internally! :)).

If a user does have that issue, I think we can assume that they are an advanced enough user to recreate their logins as needed